src/main/scala/firrtl/annotations/analysis/DuplicationHelper.scala

```diff
@@ -0,0 +1,146 @@
+        val finalName = firrtl.passes.Uniquify.findValidPrefix(prefix, Seq(postfix), ns) + postfix
+        allModules += finalName
+        cachedNames((top, path)) = finalName
+        finalName
+      case Some(newName) => newName
+    }
+  }
+
+  /** Return the duplicated module (formerly originalOfModule) instantiated by instance in newModule (formerly
+    *  originalModule)
+    * @param originalModule original encapsulating module
+    * @param newModule new name of encapsulating module
+    * @param instance instance name being declared in encapsulating module
+    * @param originalOfModule original module being instantiated in originalModule
+    * @return
+    */
+  def getNewOfModule(originalModule: String,
+                     newModule: String,
+                     instance: Instance,
+                     originalOfModule: OfModule): OfModule = {
+    dupMap.get(originalModule) match {
+      case None => // No duplication, can return originalOfModule
+        originalOfModule
+      case Some(newDupedModules) =>
+        newDupedModules.get(newModule) match {
+          case None if newModule != originalModule => throwInternalError("BAD")
+          case None => // No duplication, can return originalOfModule
+            originalOfModule
+          case Some(newDupedModule) =>
+            newDupedModule.get(instance) match {
+              case None => // Not duped, can return originalOfModule
+                originalOfModule
+              case Some(newOfModule) =>
+                newOfModule
+            }
+        }
+    }
+  }
+
+  /** Returns the names of this module's duplicated (including the original name)
+    * @param module
+    * @return
+    */
+  def getDuplicates(module: String): Set[String] = {
+    dupMap.get(module).map(_.keys.toSet[String]).getOrElse(Set.empty[String]) ++ Set(module)
+  }
+
+  /** Rewrites t with new module/instance hierarchy calculated after repeated calls to [[expandHierarchy]]
+    * @param t A target
+    * @return t rewritten, is a seq because if the t.module has been duplicated, it must now refer to multiple modules
+    */
+  def makePathless(t: IsMember): Seq[IsMember] = {
+    val top = t.module
+    val path = t.asPath
+    val newTops = getDuplicates(top)
+    newTops.map { newTop =>
+      val newPath = mutable.ArrayBuffer[TargetToken]()
+      path.foldLeft((top, newTop)) { case ((originalModule, newModule), (instance, ofModule)) =>
+        val newOfModule = getNewOfModule(originalModule, newModule, instance, ofModule)
+        newPath ++= Seq(instance, newOfModule)
+        (ofModule.value, newOfModule.value)
+      }
+      val module = if(newPath.nonEmpty) newPath.last.value.toString else newTop
+      t.notPath match {
+        case Nil => ModuleTarget(t.circuit, module)
+        case Instance(i) :: OfModule(m) :: Nil => ModuleTarget(t.circuit, module)
+        case Ref(r) :: components => ReferenceTarget(t.circuit, module, Nil, r, components)
```

Comment:
The change to Vector makes this match always fail since these all are List extractions.

Can be made generic Seq like so:
```scala
        case Seq() => ModuleTarget(t.circuit, module)
        case Instance(i) +: OfModule(m) +: Seq() => ModuleTarget(t.circuit, module)
        case Ref(r) +: components => ReferenceTarget(t.circuit, module, Nil, r, components)
```
We should probably also exercise these paths in tests

Reply:
Actually all implementations of `notPath` return lists, so the tests do in fact cover this case. However, I've still changed the `::` to `+:`.

Reply:
If all implementations of `notPath` return lists then how did I get a match error with a Vector on this? Prior to https://github.com/freechipsproject/firrtl/pull/865/commits/792472fbc733ebf441e12bfed52c04a015f458c9 at least.

Reply:
Hmm.... Well then. Not sure, let me look closer.

Reply:
Ahh so in `ReferenceTarget` I append `component`, which could be a `Vector`. Good catch. Anyways, this is fixed now.